src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -413,7 +413,19 @@ private static int getTokenType(ParseTree node) {
         final int tokenType;
 
         if (node.getChildCount() == 0) {
-            tokenType = ((TerminalNode) node).getSymbol().getType();
+            final int antlrTokenType = ((TerminalNode) node).getSymbol().getType();
+            //JavadocTokenTypes.HTML_TAG_NAME != JavadocParser.HTML_TAG_NAME.
+            //JavadocTokenTypes.EMBED_HTML_TAG_NAME != JavadocParser.EMBED_HTML_TAG_NAME
+            // See Issue #5139.
+            if (antlrTokenType == JavadocTokenTypes.HTML_TAG_NAME) {
```

Comment:
Based on discussion at https://github.com/checkstyle/checkstyle/pull/5101#issuecomment-355977976 :
Since antlr does not provide us a way to override token numbering, I think we should look into a more long term solution instead of string of `if-else` changes.
We will continue to add and re-arrange tokens, so eventually this conversion list will grow.

I think we should have either a map or array of Checkstyle tokens mapped to Antlr tokens, and use this map to replace one for the other.
Checkstyle tokens will never change values, while Antlr tokens will keep the same name while the numbering changes.

Does this sound like a solution?

Reply:
Actually the list shouldn't grow much. This situation has only presented itself because it was necessary to add a a token before another. So if an alternative to a rule in parser grammar or a token is to be added before another only then will we need to go for such an approach. I can add all the remaining tags in #3332 at once and that way only one if-else would be necessary as of now.

If and when the list starts growing should we open another issue.

Reply:
ok, we can wait then. I thought this would happen more often.